ros1_bridge/__init__.py

```diff
@@ -574,7 +574,7 @@ def determine_common_services(
                         ):
                             pairs.append((ros1_srv, ros2_srv))
 
-    for pair in pairs:
+    for pair in set(pairs):
```

Comment:
In `determine_message_pairs`, the pairs are added beforehand and then only new pairs are added to the set; otherwise the for loop continues. A similar construct is done in `determine_common_services`, but all the pairs are added and then, in your fix, the pairs object is passed to `set` to give you a unique set of pairs. 

My question is can we have a solution that follows a similar construct in `determine_message_pairs` where both functions follow a similar implementation? Both functions have similar functionality, and therefore, should have similar implementation. It will help with readibility.